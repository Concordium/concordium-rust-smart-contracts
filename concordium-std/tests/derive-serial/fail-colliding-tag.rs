//! Ensure `derive(Serial)` fails when two 'tag' attributes are colliding.
use concordium_std::*;

#[derive(Serial)]
#[concordium(repr(u8))]
enum MyTaggedEnum {
    #[concordium(tag = 42)]
    One,
    #[concordium(tag = 42)]
    Two,
}

fn main() {}
