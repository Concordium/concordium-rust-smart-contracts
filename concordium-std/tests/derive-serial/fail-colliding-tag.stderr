error: The tag of 'Two' collides with the tag of 'One'.
 --> tests/derive-serial/fail-colliding-tag.rs:9:18
  |
9 |     #[concordium(tag = 42)]
  |                  ^^^

error: The tag of 'One' collides with the tag of 'Two'.
 --> tests/derive-serial/fail-colliding-tag.rs:7:18
  |
7 |     #[concordium(tag = 42)]
  |                  ^^^
